### ISSUE TITLE
Unable to authenticate using Github Enterprise

### ISSUE THREAD

**Issue by wipeout630** [microsoft/vscode-pull-request-github#6426]
- Extension version:  v0.99.2024101604 (pre-release)
- VSCode Version:  1.94.2
- OS:  Windows 10
- Repository Clone Configuration (single repository/fork of an upstream repository):  Multi-repo, multi-fork
- Github Product (Github.com/Github Enterprise version x.x.x):  Github Enterprise 3.13.2

Steps to Reproduce:

1. Configure Enterprise Uri (https://github.mycompanydomain.com).  This is configured on the User, Remote: Ubuntu, and Workspace areas.
![Image](https://github.com/user-attachments/assets/e7404246-8cca-43e4-820f-362937ae5ddf)
2. Sign-in via the Accounts modal in VS Code
3. Sign-in page takes me to github.com
4. Click cancel on the VSCode sign-in toast message, message prompts to sign-in using local server, click yes
![Image](https://github.com/user-attachments/assets/a85bbeab-153a-4910-91e0-519c2618e4e5)
![Image](https://github.com/user-attachments/assets/08ee8204-b0f3-451b-bda8-ae1dac6d7ac3)
5. Sign-in page takes me to github.com
6. Manually change the URL in my browser to my Enterprise URL (https://github.mycompanydomain.com)
7. Authentication attempts but toast message reflects bad credentials message.

I have no problems working with git via command line, and once I clone a repo into VSCode, everything works fine.

**Comment by wipeout630** [comment]
@alexr00 I am using the accounts menu.
![Image](https://github.com/user-attachments/assets/4fec03ee-8cee-401f-a54d-e80784440817)

**Comment by wipeout630** [comment]
Works, thank you!

**Comment by andrebell** [comment]
> The issue is that we don't react to the setting `github-enterprise.uri` changing, so it would have required a reload to sign in with enterprise. The fix will be in the pre-release version of the extension on Monday.
> 
> To verify:
> 
> 1. Install the latest pre-release version of the extension.
> [...]
> 10. Verify that you now see two "Sign in with ... to user GitHub Pull Requests" in the accounts menu, and that one of them revers to the uri you set for `github-enterprise.uri`

I followed along your description. I could verify two sign in buttons in the sidepanel and two in the account menu. However, on using the "sign in with ... enterprise" button I still get the error "Sign in failed: No auth flow succeeded":

![Image](https://github.com/user-attachments/assets/fa0bd5cb-1256-4f33-aa30-963d1e88a76b)

Identifier: github.vscode-pull-request-github
Version: 0.105.2025022604

Using a PAT on the Github Enterprise Server:

> [Visual Studio Code 2024-09 (read:user repo user:email workflow)](https://svgithub01001.wago.local/settings/tokens/2223) — read:org, read:user, repo, user:email, workflow

What am I missing?

**Comment by alexr00** [comment]
@andrebell are you using your proper enterprise URI for this step?

> Set github-enterprise.uri to something. It doesn't have to be an enterprise instance, it could just be https://bing.com

I just specified `bing` for testing purposes. Are  you using github.com URI?

**Comment by andrebell** [comment]
@alexr00, yes I did set it correctly. In fact when using the `Sign in with GitHub enterprise`  button I get redirected to the server to generate a PAT

**Comment by alexr00** [comment]
@andrebell is it a github.com URI?

**Comment by TylerLeonhardt** [comment]
I am suspicious of a proxy/firewall issue.


Can you open the Command Palette (<kbd>cmd</kbd>/<kbd>ctrl</kbd>+<kbd>shift</kbd>+<kbd>p</kbd>) and run the following command:

```
>Developer: GitHub Copilot Chat Diagnostics
```

Please share the output here.

**Comment by andrebell** [comment]
@TylerLeonhardt ,

I have done so and received the following output:

> ## GitHub Copilot Chat
> 
> - Extension Version: 0.24.1 (prod)
> - VS Code: vscode/1.97.2
> - OS: Windows
> 
> ## Network
> 
> User Settings:
> ```json
>   "http.systemCertificates": false,
>   "github.copilot.advanced.debug.useElectronFetcher": true,
>   "github.copilot.advanced.debug.useNodeFetcher": false,
>   "github.copilot.advanced.debug.useNodeFetchFetcher": true
> ```
> 
> Connecting to https://api.github.com:
> - DNS ipv4 Lookup: 140.82.121.5 (19 ms)
> - DNS ipv6 Lookup: Error (2 ms): getaddrinfo ENOTFOUND api.github.com
> - Proxy URL: None (2 ms)
> - Electron fetch (configured): HTTP 403 (120 ms)
> - Node.js https: HTTP 403 (106 ms)
> - Node.js fetch: HTTP 403 (67 ms)
> - Helix fetch: HTTP 403 (361 ms)
> 
> Connecting to https://api.business.githubcopilot.com/_ping:
> - DNS ipv4 Lookup: 140.82.113.22 (157 ms)
> - DNS ipv6 Lookup: Error (23 ms): getaddrinfo ENOTFOUND api.business.githubcopilot.com
> - Proxy URL: None (4 ms)
> - Electron fetch (configured): HTTP 200 (345 ms)
> - Node.js https: HTTP 200 (348 ms)
> - Node.js fetch: HTTP 200 (331 ms)
> - Helix fetch: HTTP 200 (618 ms)
> 
> ## Documentation
> 
> In corporate networks: [Troubleshooting firewall settings for GitHub Copilot](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-firewall-settings-for-github-copilot).
>

**Comment by TylerLeonhardt** [comment]
@andrebell I've opened a new issue for you:
https://github.com/microsoft/vscode-pull-request-github/issues/6679

The right person will chime in.

**Comment by andrebell** [comment]
@alexr00 and @TylerLeonhardt , thanks for your great support. I really appreciate it.
